src/prefect/engine.py

```diff
@@ -1592,6 +1600,8 @@ async def report_flow_run_crashes(flow_run: FlowRun, client: OrionClient):
 
         # Reraise the exception
         raise exc from None
+    finally:
+        signal.signal(signal.SIGTERM, original_sigterm_handler)
```

Comment:
Sorry if you covered this already, but what if the SIGTERM is not from Prefect? SIGTERM _should_ be a crash unless the flow is in a CANCELLING state, right?

Reply:
I elected to have our own handling always be in place, but I could be convinced otherwise. If a true SIGTERM is sent, I didn't see the need to _not_ attempt a graceful shutdown. But if we were to override the SIGKILL handler I think it would be a different story.

Reply:
I think a graceful shutdown should occur either way, but should we report a CANCELLED state when it was not cancelled by Prefect?

Reply:
Perhaps we can have an orchestration rule that converts CANCELLED reports to CRASHED if the run is not in a CANCELLING state.

Reply:
I remember we discussed at some point in the past whether or not a deliberate SIGTERM should be also considered a form of cancellation, or if CANCELLED should refer explicitly to the prefect functionality—I'm leaning towards the latter because SIGTERM might come from different places?

Reply:
Is a SIGTERM that causes a graceful shutdown a Crash though?

Reply:
It'll still interrupt the user's work, right? The flow doesn't run to completion. That's a tough question. I am leaning towards the same reasoning as Dustin that a signal causes the flow to exit, when not triggered by Prefect, should not be a CANCELLED state.

Reply:
:+1: I'll raise a more generic exception if it's not in a CANCELLED state.

Reply:
I've also reached out to other stakeholders to confirm the behavior here.

Reply:
I don't think the client should check its state, I think it should just propose the CRASH and the orchestrator can say: Infrastructure teardown during cancellation is expected then REJECT that transition and ensure the run is marked as CANCELLED.

Reply:
Our states reflect Prefect's view of the world. If a flow run is stopped exogenously, if if we can presume that it was likely intentional, it should be in a CRASHED state. This is similar to Prefect not showing a flow run in a SCHEDULED if it was being triggered by cron, because it doesn't know about that schedule.

Reply:
Proposing state might be tricky there since sig handlers are sync, but I'll see what we can do.

Reply:
Crash reporting is already proposing a state — you can just raise an exception using the SIGTERM handler like you are now but not ignore it in crash handling, right?

Reply:
Yes, I thought you were saying I _shouldn't_ do that. Sorry for the confusion!